clients/roscpp/src/libros/io.cpp

```diff
@@ -44,9 +44,16 @@
 #else
   #include <cstring> // strerror
   #include <fcntl.h> // for non-blocking configuration
+  #include <sys/epoll.h>
```

Comment:
I am not opposed to any of the options. But I would think that maintaining the current pattern of keeping the implementations in the same file will likely be easier to review. If you decide to split the implementation please make sure to use separate commits for the following steps:
* copy existing file to new files without any changes
* remove redundant parts
* make the code work again from the new files

This will ensure that the git history is easy to follow.